packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -70,23 +72,39 @@ class ServerApi {
         return api.getCompleted(json.encode(true));
       case apiGetDevToolsFirstRun:
         // Has DevTools been run first time? To bring up analytics dialog.
+        //
+        // Additionally, package:unified_analytics will show a message if it
+        // is the first run with the package or the consent message version has
+        // been updated
+        final isFirstRun =
+            _devToolsUsage.isFirstRun || analytics.shouldShowMessage;
         return api.getCompleted(
-          json.encode(_devToolsUsage.isFirstRun),
+          json.encode(isFirstRun),
         );
       case apiGetDevToolsEnabled:
         // Is DevTools Analytics collection enabled?
+        final isEnabled =
+            _devToolsUsage.analyticsEnabled && analytics.telemetryEnabled;
```

Comment:
when these diverge, should we be setting the value of `_devtoolsUsage.analyticsEnabled` to `analytics.telemetryEnabled`?

Reply:
I think once we have fully migrated, we could probably get rid of `_devtoolsUsage` all together since the package will check itself when an event is attempted to be sent. Example below of how we do it in the flutter-tool by just calling the send method without any wrapping class

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/runner.dart#L190